core/src/main/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlanner.java

```diff
@@ -307,6 +309,19 @@ public CompactionPlan makePlan(PlanningParameters params) {
         group = findDataFilesToCompact(filesCopy, params.getRatio(), maxFilesToCompact,
             maxSizeToCompact);
 
+        while (!group.isEmpty() && !Collections.disjoint(group, expectedFiles)) {
+          // remove these files as compaction candidates because they include a file that a running
+          // compaction would produce
+          filesCopy.removeAll(group);
+          // Create a fake file+size entry that predicts what this projected compaction would
+          // produce
+          var futureFile = getExpectedFile(group, nextExpected);
+          Preconditions.checkState(expectedFiles.add(futureFile));
+          // look for any compaction work in the remaining set of files
+          group = findDataFilesToCompact(filesCopy, params.getRatio(), maxFilesToCompact,
+              maxSizeToCompact);
+        }
+
```

Comment:
You have a higher probability of `group` being empty at the end of this loop with these changes. Is that the intended effect?

Reply:
Yeah in most cases it probably will end up being empty.   When its empty it means no work was found that meets the compaction ratio.  When its not empty and exits the loop it means work was found that meets the compaction ratio and would not use the result of a running compaction.  This second case is more likely to occur when a tablet has built up a lot of files and has running compactions, so this will probably not be the norm.